### ISSUE TITLE
Terminal: word navigation not properly read by screen reader

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode#193459]
1. Launch Visual Studio code, and open terminal window.
2. Type the command "Git 1=diff commit" in the terminal window.
3. Start the NVDA reader.
4. Change the focus of the mouse cursor to the start of the command "Git 1=diff commit"
5. To traverse the words one by one in the above command, hit Ctrl + Right.
 

Expected output

==============

After hitting the Ctrl+Right for the first time, NVDA should read and speak "Git".
After hitting the Ctrl+Right again, NVDA should read and speak "1".
After hitting the Ctrl+Right again, NVDA should read and speak "=" and so on.
 

Actual output

==============

Hitting the Ctrl+Right multiple times should speak the words in the command , NVDA just reads and speaks "Blanks" or the first word "Git" every time we hit Ctrl +Right.

fyi @Tyriar

**Comment by meganrogge** [comment]
This currently works well in the terminal's accessible view. I'm not sure if we'll be able to support this in the terminal buffer.

**Comment by meganrogge** [comment]
@isidorn do you think we can close this given the workaround IE the terminal accessible view?

**Comment by lunasfacilitator** [comment]
Hi, I don't see this as a feature request, but rather as a bug. It shouldn't read blanks. It happens with ctrl+right and ctrl+left, but it also happens if you go character by character in either direction with right or left as well. I think the description of the bug should be made more general, that it just doesn't read anything at all, not just while holding down ctrl.

The other Microsoft terminals' ability to read what you have written isn't a feature, it's a basic necessity. And just like if the Microsoft Terminal couldn't read the characters you wrote back to you would be a bug , so is this.

Yes, there's a workaround, which is to switch to the accessible buffer and read it there, and then go back to the terminal to edit it. Please try doing this. It's unworkable. Imagine you have a long command typed in and you switch to the accessible buffer and you find a typo. Well, now you need to count what character is mistyped. Then you need to switch back to the terminal and count yet again until you get to that character and change it and hope you got it right. If it's character 60, this will be hard. Really it'll be hard even if it's character 5.

Am I missing something here? Is there a better way to do this? Please give it a shot without looking at the screen and let me know.

**Comment by meganrogge** [comment]
Hi, there was some side discussion about left and right arrow keys not working in the terminal when in screen reader mode IE not reading the characters. I cannot reproduce this. Could someone pls add this setting `"terminal.integrated.devMode": true`, and record a gif?

**Comment by isidorn** [comment]
I can reproduce. With latest VS Code Insiders and latest NVDA, when I navigate word by word in terminal buffer only "blank" is read.

https://github.com/microsoft/vscode/assets/1926584/81ba81fb-68ab-441e-8291-343901058530

**Comment by meganrogge** [comment]
@isidorn yes, I can reproduce that. It will be very challenging to solve that issue. What I cannot reproduce is an issue with just using left and right arrow keys. 

The user is suggesting we should update the title to be broader and I disagree, as I feel it's fitting ATM.

**Comment by meganrogge** [comment]
@isidorn I still stand by what I said above.

**Comment by isidorn** [comment]
Ok. Let's discuss in our sync on Monday.

**Comment by meganrogge** [comment]
This is a feature, labeling as such

**Comment by meganrogge** [comment]
On macOS, In the terminal, I type `one two three`.  with voiceOver on, I use `option + left/rightArrow` to jump between the words and VoiceOver reads them. I'll try on windows.

**Comment by meganrogge** [comment]
This also works for me on windows when using NVDA.

@isidorn looks like you're using Windows Powershell. You'll need to use an updated version of powershell for this to work, as it relies on shell integration to sync the text area.

@lunasfacilitator can you pls try in a terminal with shell integration? Do you know what version of pwsh you're using?

**Comment by lunasfacilitator** [comment]
Hi, terminal.integrated.devMode fixed the problem of the terminal being completely silent. Thanks! Can you set it automatically when screen reader mode is enabled for the rest of the folks out there that don't know about this setting?

So, re word navigation, I typed in:
ls -al blah

I hit the home key and hit ctrl+right a few times and it said "ls", then "al". It didn't mention the "-" or "blah".

**Comment by meganrogge** [comment]
`terminal.integrated.devMode` shouldn't have fixed anything 🤔 . You should be able to disable that. 

Your steps work for me.

**Comment by lunasfacilitator** [comment]
Navdeep called me and over zoom we changed some setting, surely that one, and then for the first time I was able to go char by char with left and right and actually read chars that I typed into the terminal. This is the only way to make sure what you typed is correct before you run it. So, yes, that setting seemed to make the difference. If I'm right about this, it should be enabled by default for screen reader users. Again, if this doesn't work, the whole terminal is broken, whereas word navigation is not as serious.

Re word navigation, yes, the "-" not being read is probably the screen reader like you said, but it should have read "blah".
